src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -1,643 +1,873 @@
-      // * Inflate bounding boxes by 33% of longest dimension
-      //   to avoid zero-width dimensions
-      // * Find the average extents (range) of the boxes
-      //   Assumption is that elements are roughly the same size
-      // * Grid resolution for each dimension is overall box width
-      //   divided by half the average element width
-      SpaceVec ranges;
-      for(int i=0; i< m_meshWrapper1.numElems(); ++i)
-      {
-         auto& bbox = m_meshBBoxes1[i];
-         inflateBBox(bbox);
+    ranges /= m_mesh1.numberOfElements();
 
-         ranges += bbox.range();
+    // Compute grid resolution from average bbox size
+    typename ImplicitGridType::GridCell resolution;
+    SpaceVec bboxRange = m_gridBBox.range();
+    const RealT scaleFac = 0.5; // TODO is this mesh dependent?
+    for(int i=0; i < D; ++i)
+    {
+      resolution[i] = static_cast<IndexT>(
+            std::ceil( scaleFac * bboxRange[i] / ranges[i] ));
+    }
 
-         // build up overall bounding box along the way
-         m_gridBBox.addBox( bbox );
-      }
+    // Next, initialize the ImplicitGrid
+    m_grid.initialize( m_gridBBox, &resolution, m_mesh1.numberOfElements());
 
-      // inflate grid box slightly so elem bounding boxes are not on grid bdry
-      m_gridBBox.scale(1 + bboxTolerance);
+    // Finally, insert the elements
+    for(int i=0; i< m_mesh1.numberOfElements(); ++i)
+    {
+      m_grid.insert(m_meshBBoxes1[i], i);
+    }
 
-      ranges /= m_meshWrapper1.numElems();
+    // Output some info for debugging
+    if(true)
+    {
+      SLIC_INFO("Implicit Grid info: "
+          << "\n Mesh 1 bounding box (inflated): " << m_gridBBox
+          << "\n Avg range: " << ranges
+          << "\n Computed resolution: " << resolution );
 
-      // Compute grid resolution from average bbox size
-      typename ImplicitGridType::GridCell resolution;
-      SpaceVec bboxRange = m_gridBBox.range();
-      const real scaleFac = 0.5; // TODO is this mesh dependent?
-      for(int i=0; i < D; ++i)
+      SpatialBoundingBox bbox2;
+      for(int i=0; i< m_mesh2.numberOfElements(); ++i)
       {
-         resolution[i] = static_cast<IndexType>(
-               std::ceil( scaleFac * bboxRange[i] / ranges[i] ));
+        bbox2.addBox( elementBoundingBox(m_mesh2, i) );
       }
 
-      // Next, initialize the ImplicitGrid
-      m_grid.initialize( m_gridBBox, &resolution, m_meshWrapper1.numElems());
+      SLIC_INFO( "Mesh 2 bounding box is: " << bbox2 );
+    }
+  }; // end initialize()
+
+  /*!
+   * Use the spatial index to find candidates in first mesh for each
+   * element in second mesh of coupling scheme.
+   */
+  void findInterfacePairs() override
+  {
+    using BitsetType = typename ImplicitGridType::BitsetType;
+
+    // Extract some mesh metadata from coupling scheme
+    auto& mesh1 = m_coupling_scheme->getMesh1();
+    auto& mesh2 = m_coupling_scheme->getMesh2();
+    auto& contactPairs = m_coupling_scheme->getInterfacePairs();
```

Comment:
It looks like the interface pairs was cleared in `initialize()`. Do we need to reserve space for this similar to what was done in the first `findInterfacePairs()` above (in search base class?)?